lib/possible_bipartition.rb

```diff
@@ -1,4 +1,85 @@
+#breath-first
+def puppy_checker(node, pink, blue)
+  if pink.include?(node)
+    return "pink"
+  elsif blue.include?(node)
+    return "blue"
+  else
+    return "uncolored"
+  end
+end
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  pink = Set.new
+  blue = Set.new 
+  
+  dislikes.each_with_index do |adjacent_nodes , i|
+    node_color = puppy_checker(i, pink, blue)
+    if node_color == "pink"
+      pink << i 
+    else
+      blue << i
+      node_color = "blue"
+    end 
+    
+    adjacent_nodes.each do |j|
+      adjacent_color = puppy_checker(j, pink, blue)
+      return false if node_color == adjacent_color
+      if node_color == "pink"
+        blue << j
+      else
+        pink << j
+      end
+    end
+  end 
```

Comment:
Shouldn't this section then add the adjacent nodes to `j` to a queue or stack and continue until the queue or stack is empty?  

You're not hitting the nodes in the order they would be encountered in a traversal.  Instead you are going through them in order.